### PR TITLE
[No QA] Simplify electron-builder configs

### PR DESCRIPTION
**Pull request by roryabraham**
### Details
Following up on https://github.com/Expensify/App/pull/7665 to do some cleanup.

### Fixed Issues
n/a

### Tests
1. Run `npm run desktop`, and verify that the app loads correctly.
1. Make a visible change to a React Native file, then save. The desktop app should hot-reload.
1. Run `npm run desktop-build-local`, install the app and make sure it works! It should have a production app icon, not a staging app icon.
1. Run `npm run desktop-build-staging-local`, install the app and make sure it works! It should have a staging app icon, not a production app icon.
1. Merge this PR and pray that it doesn't break our desktop deploys 🙃 

- [x] Verify that no errors appear in the JS console

### QA Steps
None.

- [ ] Verify that no errors appear in the JS console

### Tested On

- [ ] Web
- [ ] Mobile Web
- [x] Desktop
- [ ] iOS
- [ ] Android

#### Desktop (production build)
image
image
image

#### Desktop (staging build)
image

### CODE REVIEW

**Comment by botify** [comment]
npm has a `package.json` file and a `package-lock.json` file. It seems you updated one without the other, which is usually a sign of a mistake. If you are updating a package make sure that you update the version in package.json then run `npm install`

**Comment by botify** [comment]
npm has a `package.json` file and a `package-lock.json` file. It seems you updated one without the other, which is usually a sign of a mistake. If you are updating a package make sure that you update the version in package.json then run `npm install`

**Comment by botify** [comment]
npm has a `package.json` file and a `package-lock.json` file. It seems you updated one without the other, which is usually a sign of a mistake. If you are updating a package make sure that you update the version in package.json then run `npm install`

**Comment by botify** [comment]
npm has a `package.json` file and a `package-lock.json` file. It seems you updated one without the other, which is usually a sign of a mistake. If you are updating a package make sure that you update the version in package.json then run `npm install`

**Comment by roryabraham** [comment]
Wait, _do_ I need an `npm i`?

**Comment by roryabraham** [comment]
No, no I do not.

**Comment by roryabraham** [comment]
Interesting side-effect, is this supposed to be there?

image

**Comment by roryabraham** [comment]
No, that shouldn't be there...

**Comment by neil-marcellini** [comment]
I'm going to un-assign from this review because I've been swamped by PRs lately and this one is quite a bit over my head.

File: config/electronBuilder/electronBuilder.ghactions.config.js
Comment (tgolen) [inline config/electronBuilder/electronBuilder.ghactions.config.js:9]:
```suggestion
        bucket: process.env.NODE_ENV === 'staging' ? 'staging-expensify-cash' : 'expensify-cash',
```
I think this is fine, and the extra variable really isn't necessary.

File: config/electronBuilder/electronBuilder.local.config.js
Comment (tgolen) [inline config/electronBuilder/electronBuilder.local.config.js:3]:
```suggestion
const isStaging = process.env.NODE_ENV === 'staging';
```
"build" just doesn't really need to be in the name, and it's really not a build but more of an environment.

File: package.json
Comment (tgolen) [inline package.json:16]: Can you please move this down so that it's directly above the other desktop builds?

**Review by tgolen — CHANGES_REQUESTED** [review]
Looking really great! A few suggestions

**Comment by botify** [comment]
npm has a `package.json` file and a `package-lock.json` file. It seems you updated one without the other, which is usually a sign of a mistake. If you are updating a package make sure that you update the version in package.json then run `npm install`

**Comment by roryabraham** [comment]
Updated!

**Comment by botify** [comment]
npm has a `package.json` file and a `package-lock.json` file. It seems you updated one without the other, which is usually a sign of a mistake. If you are updating a package make sure that you update the version in package.json then run `npm install`

**Comment by botify** [comment]
npm has a `package.json` file and a `package-lock.json` file. It seems you updated one without the other, which is usually a sign of a mistake. If you are updating a package make sure that you update the version in package.json then run `npm install`

**Comment by roryabraham** [comment]
Taking this off draft and marking it as ready for review, but I think we should keep it on hold until it's not DBSF.

**Review by tgolen — APPROVED** [review]
Cool, this looks good to me. How come it's on HOLD?

**Comment by tgolen** [comment]
Oh, because it was Friday, got it. You can self merge this when you are ready :D

**Comment by botify** [comment]
npm has a `package.json` file and a `package-lock.json` file. It seems you updated one without the other, which is usually a sign of a mistake. If you are updating a package make sure that you update the version in package.json then run `npm install`

File: config/electronBuilder/electronBuilder.ghactions.config.js
Comment (kidroca) [inline config/electronBuilder/electronBuilder.ghactions.config.js:10]: Were there any problems when this was part of the single config?
I have no problems building local builds without this change?

E.g. I don't use `--publish` and this just creates a local build for me, no need for `desktop-build-local` and `desktop-build-local-staging` scripts

---

Separate question: If `notarize.js` is something only used for config/build signing why keep it in `desktop` ? it gets packed to the electron app due to the `desktop/*.js` pattern

File: package.json
Comment (kidroca) [inline package.json:24]: This looks a bit odd - none of the other build scripts web/mobile have to set ENV vars in their package script.
It might be better to extract a script like `desktop/start.js` but perhaps `desktop/build.js` where the appropriate `dotenv` file can be loaded. The least it would hide these details and make the scripts shorter

**Review by kidroca — COMMENTED** [review]
I'm working on reducing desktop size here: https://github.com/Expensify/App/pull/7744 and noticed some merge conflicts.
I decided to leave some notes

Do we expect any more Desktop changes, because I'm having to change my code each time a change is made?

_(managed to reduce the installer to ~90Mb)_

File: config/electronBuilder/electronBuilder.ghactions.config.js
Comment (roryabraham) [reply config/electronBuilder/electronBuilder.ghactions.config.js:10]: > Were there any problems when this was part of the single config?

I always run into issues with the notarizing step. You don't experience that?

> Separate question: If notarize.js is something only used for config/build signing why keep it in desktop ?

Just because it's specific to the desktop application.

**Comment by roryabraham** [comment]
> Do we expect any more Desktop changes

I don't expect any more for now.

**Pull request merged**
